types/webgl2/index.d.ts

```diff
@@ -385,10 +385,9 @@ interface WebGL2RenderingContext extends WebGLRenderingContext {
         x: number, y: number, width: number, height: number): void;
 
     compressedTexImage2D(target: number, level: number, internalformat: number, width: number,
-        height: number, border: number, imageSize: number, offset: number): void;
-    compressedTexImage2D(target: number, level: number, internalformat: number, width: number,
-        height: number, border: number, srcData: ArrayBufferView,
-        srcOffset?: number, srcLengthOverride?: number): void;
+        height: number, border: number, data: ArrayBufferView | DataView | null): void;
```

Comment:
Lines 388-391 should stay unchanged. Those are additional overloads available in WebGL2 for compressedTexImage2D [see [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/compressedTexImage2D)]

Unrelated: Is the separate listing of `ArrayBufferView` and `DataView` intentional? `DataView` inherits from `ArrayBufferView`, right?

Reply:
The spec suggests this is a `void*` which technically allows for null. 

The `DataView` part should be removed, since as you noted, it is already a part of `ArrayBufferView`. that was an oversight on my part.

Reply:
Right, the `null` part is fine.
The problem is that your commit would replace the two newly introduced WebGL 2 overloads for `compressedTexImage2D` with old (WebGL 1.0) `compressedTexImage2D` and `compressedTexSubImage2D` functions. Those are of course still valid, but already covered in [lib.es6.d.ts](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es6.d.ts#L18321-L18322)

Reply:
sorry, that was a copy/past mistake. i fixed that.

Reply:
I just reviewed the spec at Khronos - it seems to be a bit inconsistent about whether/when `null` is acceptable or not. For the definitions here, I used the [IDL](https://www.khronos.org/registry/webgl/specs/latest/2.0/webgl2.idl) file.
As an example, for `void texSubImage3D` they are specifying `ArrayBufferView?` as type for `srcData`, which I translated as `ArrayBufferView|null`.
In the cases modified by this commit here, they only specified `ArrayBufferView` without `?`. I am unsure if that is intentional, or an oversight in the specification? You mentioned they accept `void*` as input - where can I find that version?